### ISSUE TITLE
value cannot be converted to type at::Half without overflow

### ISSUE THREAD

**Issue by williamlzw** [dotnet/TorchSharp#1135]
var mask = torch.ones_like(weight).triu(1).to(torch.@bool);
weight.masked_fill_(mask, -torch.finfo(torch.float32).max);//System.Runtime.InteropServices.ExternalException:“value cannot be converted to type at::Half without overflow”

//python
mask = torch.ones_like(weight).triu(1).to(torch.bool)
weight.masked_fill_(mask, -torch.inf)

**Comment by NiklasGustafsson** [comment]
@williamlzw -- I don't believe finfo(float32).max is the same as 'infinity'

Try:

```C#
weight.masked_fill_(mask, System.Single.NegativeInfinity);
```

**Comment by williamlzw** [comment]
Problem solved.
